stackslib/src/chainstate/stacks/boot/pox-4.clar

```diff
@@ -1300,6 +1324,26 @@
       (ok { stacker: stacker,
             unlock-burn-height: new-unlock-ht }))))
 
+;; Add an authorization for a signer key.
+;; When an authorization is added, the `signer-sig` argument is not required
+;; in the functions that use it as an argument.
+;; The `allowed` flag can be used to either enable or disable the authorization.
+;; Only the Stacks principal associated with `signer-key` can call this function.
+;; *New in Stacks 3.0*
+(define-public (set-signer-key-authorization (pox-addr { version: (buff 1), hashbytes: (buff 32)})
+                                             (period uint)
+                                             (reward-cycle uint)
+                                             (topic (string-ascii 12))
+                                             (signer-key (buff 33))
+                                             (allowed bool))
```

Comment:
An interesting consequence of this authorization table is that the signer has to perform this contract-call for each and every reward cycle they want to authorize.  This can get expensive (lots of contract-calls) and/or tedious (lots of transactions).  Can this code instead be written such that the caller can specify a _range_ of reward cycles?  Or perhaps an optional last reward cycle?  Then, they'd only need to update this map whenever they want to change the times for which this key is valid.

Reply:
I think this contract can just punt that optimization to the caller -- they're perfectly capable of just calling this function multiple times in a single transaction through `contract-call`. That way this contract doesn't have to deal with either implementing the iteration or implementing a range authorization map.

Reply:
Having a loop over pox contract calls is kind of expensive.